packages/netsuite-adapter/test/client/client.test.ts

```diff
@@ -135,6 +136,100 @@ describe('NetsuiteClient', () => {
+        const manifestErrorMessage = 'Details: The manifest contains a dependency on customworkflow1.workflowstate17.workflowaction33'
+        const manifestValidationError = new ManifestValidationError(manifestErrorMessage, ['customworkflow1.workflowstate17.workflowaction33'])
+        mockSdfDeploy.mockRejectedValueOnce(manifestValidationError)
+        const successChange = toChange({
+          after: new InstanceElement('instance', type, { scriptid: 'someObject' }),
+        })
+        const failedChange = toChange({
+          after: new InstanceElement('failedInstance', type, { scriptid: 'scriptid', bad_ref: '[scriptid=customworkflow1.workflowstate17.workflowaction33]' }),
+        })
+        expect(await client.deploy(
+          [successChange, failedChange],
+          SDF_CHANGE_GROUP_ID,
+          ...deployParams
+        )).toEqual({
+          errors: [manifestValidationError],
+          appliedChanges: [successChange],
+        })
+        expect(mockSdfDeploy).toHaveBeenCalledTimes(2)
+      })
+
+      it('should not apply any changes if failed scriptid cant be extracted from error message', async () => {
+        const type = new ObjectType({ elemID: new ElemID(NETSUITE, 'type') })
+        const manifestErrorMessage = 'Details: The manifest contains a dependency on some_id'
+        const manifestValidationError = new ManifestValidationError(manifestErrorMessage, ['some_id'])
+        mockSdfDeploy.mockRejectedValueOnce(manifestValidationError)
+        const successChange = toChange({
+          after: new InstanceElement('instance', type, { scriptid: 'someObject', ref: '[scriptid=some_ref]' }),
+        })
+        const failedChange = toChange({
+          after: new InstanceElement('failedInstance', type, { scriptid: 'scriptid', bad_ref: '[scriptid=failed_scriptid]' }),
+        })
+        expect(await client.deploy(
+          [successChange, failedChange],
+          SDF_CHANGE_GROUP_ID,
+          ...deployParams
+        )).toEqual({
+          errors: [manifestValidationError],
+          appliedChanges: [],
+        })
+        expect(mockSdfDeploy).toHaveBeenCalledTimes(1)
+      })
+
+      it('should try to deploy again after ManifestValidationError from salto reference', async () => {
+        const type = new ObjectType({ elemID: new ElemID(NETSUITE, 'center') })
+        const manifestErrorMessage = 'Details: The manifest contains a dependency on custcenter1'
+        const manifestValidationError = new ManifestValidationError(manifestErrorMessage, ['custcenter1'])
+        mockSdfDeploy.mockRejectedValueOnce(manifestValidationError)
+        const successChange = toChange({
+          after: new InstanceElement('instance', type, { scriptid: 'someObject' }),
+        })
+        const centerInstance = new InstanceElement('name', type, { scriptid: 'custcenter1' })
+        const saltoRefExp = new ReferenceExpression(
+          centerInstance.elemID.createNestedID('instance', 'custcenter1', SCRIPT_ID),
```

Comment:
centerInstance.elemID is `netsuite.center.instance.custcenter1`
```suggestion
          centerInstance.elemID.createNestedID(SCRIPT_ID),
```